### DISCUSSION TITLE
WebSocket connection failed

### DISCUSSION THREAD

**Discussion by okontsograda** [outline/outline#3893] [Self Hosting]
I'm setting up outline wiki on a production environemtn and I've come across this websocket issue. I'm able to login through the slack configuration and create a new document, but when I'm unable to get any options to add content to the document. Would this be something with needing to upgrade all of my connections to https, or is this a domain related issue? I never experienced this when building it locally and running it off localhost.

`WebSocket connection to 'wss://wiki.mydomain.com/realtime/?EIO=3&transport=websocket' failed: `

This error is being spammed in the console of the browser.

**Comment by tommoor** [comment]
> Would this be something with needing to upgrade all of my connections to https

HTTPS is required in production.

**Reply by okontsograda** [reply]
Would I do this through a proxy and upgrade all connections that way? What would be the ideal way to go about getting this resolved?

**Comment by apoorv-mishra** [comment]
Check if #2222 helps.
